src/usbhost/interface-usbhost.cpp

```diff
@@ -0,0 +1,52 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2015 Intel Corporation. All Rights Reserved.
+
+#ifdef RS2_USE_ANDROID_BACKEND
+
+#include <cstdint>
+#include <vector>
+#include "interface-usbhost.h"
+#include "messenger-usbhost.h"
+
+#include "../types.h"
+#include "usbhost.h"
+
+namespace librealsense
+{
+    namespace platform
+    {
+        usb_interface_usbhost::usb_interface_usbhost(::usb_interface_descriptor desc, ::usb_descriptor_iter it) :
+                _desc(desc)
+        {
+            for (int e = 0; e < desc.bNumEndpoints;) {
+                usb_descriptor_header *h = usb_descriptor_iter_next(&it);
+                if (h->bDescriptorType == (USB_DT_ENDPOINT & ~USB_TYPE_MASK)) {
```

Comment:
If that condition fails - does it guaranteed break on  `usb_descriptor_header *h = usb_descriptor_iter_next(&it);` ?
To prevent entering endless loop

Reply:
adding                 
if(h == NULL)
                    break;